### ISSUE TITLE
Engine panics with `segmentation violation code=0x1`

### ISSUE THREAD

**Issue by masih** [filecoin-project/go-indexer-core#80]
After [upgrading `x/sys`](https://github.com/filecoin-project/go-indexer-core/pull/79) dependency panic still occurs on put when values retrieved from cache are checked for equality.

```
[signal SIGSEGV: segmentation violation code=0x1 addr=0x7f7f772e6999 pc=0x410af1]

goroutine 275 [running]:
runtime.throw({0x15d88b4?, 0xc00536a000?})
	/usr/local/go/src/runtime/panic.go:992 +0x71 fp=0xc00903e898 sp=0xc00903e868 pc=0x4473b1
runtime.sigpanic()
	/usr/local/go/src/runtime/signal_unix.go:825 +0x305 fp=0xc00903e8e8 sp=0xc00903e898 pc=0x45d785
memeqbody()
	/usr/local/go/src/internal/bytealg/equal_amd64.s:108 +0xd1 fp=0xc00903e8f0 sp=0xc00903e8e8 pc=0x410af1
bytes.Equal(...)
	/usr/local/go/src/bytes/bytes.go:20
github.com/filecoin-project/go-indexer-core.Value.Match(...)
	/go/pkg/mod/github.com/filecoin-project/go-indexer-core@v0.6.0/value.go:58
github.com/filecoin-project/go-indexer-core.Value.Equal(...)
	/go/pkg/mod/github.com/filecoin-project/go-indexer-core@v0.6.0/value.go:63
github.com/filecoin-project/go-indexer-core/engine.(*Engine).Put(0xc002dc0b00, {{0xc0103faed0, 0x22}, {0xc013080d40, 0x3d, 0x3f}, {0xc002fe2db8, 0x2, 0x2}}, {0xc00a1f4000, ...})
	/go/pkg/mod/github.com/filecoin-project/go-indexer-core@v0.6.0/engine/engine.go:93 +0x645 fp=0xc00903eae8 sp=0xc00903e8f0 pc=0xa28945
github.com/filecoin-project/storetheindex/internal/ingest.(*Ingester).indexAdMultihashes(0xc00052c3c0, {{0x1a54eb0, 0xc0009c1060}, {0xc013080f00, 0x3b}, {0xc0009c1240, 0x1, 0x1}, {0xc009343400, 0x272, ...}, ...}, ...)
	/storetheindex/internal/ingest/linksystem.go:534 +0x705 fp=0xc00903ed18 sp=0xc00903eae8 pc=0xde7765
github.com/filecoin-project/storetheindex/internal/ingest.(*Ingester).ingestEntryChunk(0xc00052c3c0, {0x1a55828?, 0xc0084d8660?}, {{0x1a54eb0, 0xc0009c1060}, {0xc013080f00, 0x3b}, {0xc0009c1240, 0x1, 0x1}, ...}, ...)
	/storetheindex/internal/ingest/linksystem.go:475 +0x13e fp=0xc00903ee50 sp=0xc00903ed18 pc=0xde6d7e
```

**Comment by gammazero** [comment]
Line 58 of `value.go`  is
```go
return v.ProviderID == other.ProviderID && bytes.Equal(v.ContextID, other.ContextID)
```
So `bytes.Equal` fails on bad contextID.

**Comment by masih** [comment]
Two nodes paniced this morning because of this; going to pick this up.

**Comment by masih** [comment]
```
	/go/pkg/mod/github.com/filecoin-project/go-indexer-core@v0.6.6/engine/engine.go:93 +0x64e fp=0xc0006a0ae8 sp=0xc0006a08f0 pc=0xa3b10e
github.com/filecoin-project/go-indexer-core/engine.(*Engine).Put(0xc002d35d40, {{0xc005dba390, 0x22}, {0xc004ea15c0, 0x3d, 0x3f}, {0xc0012d5ba0, 0x2, 0x2}}, {0xc00544e000, ...})
	/go/pkg/mod/github.com/filecoin-project/go-indexer-core@v0.6.6/value.go:72
github.com/filecoin-project/go-indexer-core.Value.Equal(...)
	/go/pkg/mod/github.com/filecoin-project/go-indexer-core@v0.6.6/value.go:67
github.com/filecoin-project/go-indexer-core.Value.Match(...)
	/usr/local/go/src/bytes/bytes.go:20
bytes.Equal(...)
	/usr/local/go/src/internal/bytealg/equal_amd64.s:108 +0xd1 fp=0xc0006a08f0 sp=0xc0006a08e8 pc=0x40d811
memeqbody()
	/usr/local/go/src/runtime/signal_unix.go:842 +0x2c5 fp=0xc0006a08e8 sp=0xc0006a0898 pc=0x45bf85
runtime.sigpanic()
	/usr/local/go/src/runtime/panic.go:1047 +0x5d fp=0xc0006a0898 sp=0xc0006a0868 pc=0x4455fd
runtime.throw({0x162117d?, 0xc000827000?})
```

**Comment by masih** [comment]
Resolved by #95
